### ISSUE TITLE
[Feature Request]: Sub patch to be activated for a moment at disabling their parent patch

### ISSUE THREAD

**Issue by Berylskid** [PCSX2/pcsx2#9008]
### Description

This is to make it easier to revert/restore patched memory without additional patches in per-game setting window.

### Reason

This allows user to deactivate effects of patches just by unchecking them in per-game setting instead of rebooting game or checking other disabling-patches in per-game setting.

### Examples

Here's an example of pnach file:
```ini
[patch]
pnach=1,EE,00246680,extended,00000040 //20 by default

[patch\disable]
pnach=1,EE,00246680,extended,00000020
ondisable=1
```
- `[patch\disable]` is hidden from per-game setting window.
- `[patch\disable]` will be automatically activated for a moment at disabling `[patch]`.

**Comment by refractionpcsx2** [comment]
Nah, that would suck, I'd rather we backed up memory and restored it if the patches get disabled, but I don't think adding the new system would be overly trivial, plus that wouldn't work with savestates if you changed the enabledness/availability of the patch in between.  Doing it in patches doesn't feel right, though. Seems like unnecessary bulk, IMO.

**Comment by stenzek** [comment]
Yeah, I feel like backing up the value is probably a better option here.

But we'd need a new type, because you only want to back up code, not data.
